mlos_bench/mlos_bench/storage/sql/trial.py

```diff
@@ -43,12 +46,60 @@ def __init__(  # pylint: disable=too-many-arguments
             experiment_id=experiment_id,
             trial_id=trial_id,
             tunable_config_id=config_id,
+            trial_runner_id=trial_runner_id,
             opt_targets=opt_targets,
             config=config,
         )
         self._engine = engine
         self._schema = schema
 
+    def set_trial_runner(self, trial_runner_id: int) -> int:
+        trial_runner_id = super().set_trial_runner(trial_runner_id)
+        with self._engine.begin() as conn:
+            conn.execute(
+                self._schema.trial.update()
+                .where(
+                    self._schema.trial.c.exp_id == self._experiment_id,
+                    self._schema.trial.c.trial_id == self._trial_id,
+                    (
+                        or_(
+                            self._schema.trial.c.trial_runner_id.is_(None),
+                            self._schema.trial.c.status == str(Status.PENDING),
```

Comment:
that will produce a string `'Status.PENDING'`. Did you mean
```suggestion
                            self._schema.trial.c.status == Status.PENDING.name,
```
?

(Note to self: use `Status.*.name` everywhere in storage instead of string literals)

(Note 2: need to check for such things in unit tests)

Reply:
Yep, good call. Will add a test for that tomorrow. Thanks!

Reply:
Will fix this more broadly in a new PR.

Reply:
#920